cfgov/agreements/metadata.py

```diff
@@ -0,0 +1,33 @@
+from django.conf import settings
+
+import boto3
+from botocore.exceptions import ClientError, ConnectionError
+
+
+class Metadata:
+    def __init__(self):
+        s3 = boto3.resource("s3")
+        self.bucket = s3.Bucket(settings.AWS_STORAGE_BUCKET_NAME or "")
+        self.prefix = "a/assets/bulk_agreements/"
+        self.flexibilities = [
+            "Q1-2020",
+            "Q2-2020",
+            "Q3-2020",
+            "Q4-2020",
+            "Q1-2021",
+        ]
+        self.notes = {
+            "Q2-2019": "Agreements are incomplete due to technical submission"
+            " issues at the Bureau"
+        }
+
+    def get_objects_by_prefix(self):
+        return self.bucket.objects.filter(Prefix=self.prefix)
+
+    def get_sorted_agreements(self, reverse=True):
+        try:
+            agreements = [f.key for f in self.get_objects_by_prefix()]
+            agreements.sort(reverse=reverse)
+            return agreements
+        except (ClientError, ConnectionError):
```

Comment:
I want the error to bubble up if it's our fault, so we can fix it. If `ClientError` gets thrown when credentials are bad, won't `except ClientError` prevent us from fixing our own mistake?

Reply:
In this instance, possibly, but I think I'd prefer to suppress the nasty error page. If AWS creds are bad, jenkins jobs will be failing, smoke tests will be failing, etc. so I don't think swallowing a clienterror here would prevent us catching that.